src/common/BaseLightAccount.sol

```diff
@@ -75,6 +76,39 @@ abstract contract BaseLightAccount is BaseAccount, TokenCallbackHandler, UUPSUpg
         }
     }
 
+    /// @notice Creates a contract, this can only be called by this account.
+    /// @param initCode The initCode to deploy. NOTE: This could be replaced with transient storage in the near future,
+    /// depending on gas savings, if any.
+    function create(bytes calldata initCode, uint256 value) external payable virtual onlyAuthorized {
+        assembly ("memory-safe") {
+            // Copy the initCode to memory, then deploy the contract
+            let len := initCode.length
+            calldatacopy(0, initCode.offset, len)
+            let succ := create(value, 0, len)
+
+            // If the creation fails, revert
+            if iszero(succ) {
+                mstore(0, 0x7e16b8cd) // CreateFailed()
+                revert(28, 4)
+            }
```

Comment:
For this and the other assembly revert, I think we could do it in solidity instead of yul if we target `0.8.26`, the optimizer was [just updated](https://soliditylang.org/blog/2024/05/21/solidity-0.8.26-release-announcement#:~:text=Optimization%20for%20reverts%20with%20errors%20of%20small%20static%20encoding%20size) to include this.

But if we want to keep the target floating at a lower version with the `^` char, then yeah we can keep this.

Reply:
I think even the standard if statement revert would compile down to the same, but I don't really see a reason to jump out of assembly (and thus declare the variable outside the asm block). Fine with either solution though!